var/spack/repos/builtin/packages/gaussian/package.py

```diff
@@ -14,46 +15,261 @@ class Gaussian(Package):
 
     maintainers = ['antoniokaust']
 
+    version('16-C.01', sha256='c9eb73a9df5ca8705fcf2d7ce2d5f9aceb05ae663689f54c0a581c9d4d44fffb')
     version('16-B.01', sha256='0b2cf60aa85d2c8c8e7547446e60e8e8cb67eec20e5f13c4a3e4e7616dcdf122')
     version('09-D.01', sha256='ef14885b5e334b6ec44a93bfd7225c634247dc946416af3087ab055bf05f54cd')
 
-    @property
-    def ver(self):
-        return self.version.string.split('-')[0]
+    depends_on('tcsh', type='build')
+
+    conflicts('%cray', msg='Gaussian can only be built with the PGI compiler')
+    conflicts('%gcc', msg='Gaussian can only be built with the PGI compiler')
+    conflicts('%intel', msg='Gaussian can only be built with the PGI compiler')
+    conflicts('%nag', msg='Gaussian can only be built with the PGI compiler')
+    conflicts('%xl', msg='Gaussian can only be built with the PGI compiler')
 
     @property
-    def g_root(self):
-        return join_path(self.prefix, 'g' + self.ver)
+    def g_name(self):
+        return 'g{0}'.format(self.version.up_to(1))
 
     @property
-    def g_bsd(self):
-        return join_path(self.g_root, 'bsd')
+    def g_root(self):
+        return self.prefix.join(self.g_name)
 
     def url_for_version(self, version):
         return "file://{0}/g{1}.tgz".format(os.getcwd(), version)
 
     def install(self, spec, prefix):
-        install_tree('.', self.g_root)
+        # Spacks strips the single dir inside the tarball, but Gaussian
+        # needs it -> move them back
+        files = os.listdir()
+        mkdirp(self.g_name)
+        for f in files:
+            os.rename(f, join_path(self.g_name, f))
+
+        with working_dir(self.g_name):
+            # fix the shebang to use our tcsh
+            filter_file('/bin/csh -(fx|xf)', '/usr/bin/env tcsh\nset echo',
+                'bsd/bldg16',
+                'bsd/l4x',
+                'bsd/do-util',
+                'bsd/make-nutill',
+                'bsd/mkt',
+                'bsd/rfhello',
+                'bsd/ssrunx',
+                'bsd/testunf',
+                'bsd/upd-util',
+                string=True,
+                backup=False)
+
+            filter_file('/bin/csh -f', '/usr/bin/env tcsh',
```

Comment:
You dropped the `-f` on purpose?

Reply:
yes and no: I assume that the environment used by Spack is clean enough, and when using `/usr/bin/env ...` you can't pass additional arguments.